packages/yew/src/dom_bundle/btext.rs

```diff
@@ -89,6 +89,66 @@ impl std::fmt::Debug for BText {
     }
 }
 
+#[cfg(feature = "hydration")]
+mod feat_hydration {
+    use super::*;
+
+    use web_sys::Node;
+
+    use crate::dom_bundle::{Fragment, Hydratable};
+    use wasm_bindgen::JsCast;
+
+    impl Hydratable for VText {
+        fn hydrate(
+            self,
+            root: &BSubtree,
+            parent_scope: &AnyScope,
+            parent: &Element,
+            fragment: &mut Fragment,
+        ) -> (NodeRef, Self::Bundle) {
+            if let Some(m) = fragment.front().cloned() {
+                // better safe than sorry.
+                if m.node_type() == Node::TEXT_NODE {
+                    if let Ok(m) = m.dyn_into::<TextNode>() {
+                        // pop current node.
+                        fragment.pop_front();
+
+                        // TODO: It may make sense to assert the text content in the text node against
+                        // the VText when #[cfg(debug_assertions)] is true, but this may be complicated.
+                        // We always replace the text value for now.
+                        //
+                        // Please see the next comment for a detailed explanation.
+                        m.set_node_value(Some(self.text.as_ref()));
+
+                        return (
+                            NodeRef::new(m.clone().into()),
+                            BText {
+                                text: self.text,
+                                text_node: m,
+                            },
+                        );
+                    }
+                }
+            }
+
+            // If there are multiple text nodes placed back-to-back in SSR, it may be parsed as a single
+            // text node by browser, hence we need to add extra text nodes here if the next node is not a text node.
+            // Similarly, the value of the text node may be a combination of multiple VText vnodes.
+            // So we always need to override their values.
```

Comment:
> When vtext gets rendered to string, it could insert additional comment nodes around it

This wouldn't work for tags like `<style>` or `<script>`.

> Should also make it way less risky to call trim_start_text_nodes all over the place to "fix whitespace".

You cannot prevent whitespace before the rendered content. (Users may choose to include a blank line between `<body>` and the content, etc.)

Reply:
> This wouldn't work for tags like <style> or <script>.

I understand. Since everything is immediately rendered again, anyway, I suppose the current behaviour of confusing multiple successive text nodes wouldn't impact the result on screen.